.github/scripts/git_managers.py

```diff
@@ -0,0 +1,142 @@
+import logging
+from multiprocessing.managers import Value
+
+from github import GithubException, InputGitTreeElement
+from github.GitCommit import GitCommit
+from github.GitRef import GitRef
+from github.Issue import Issue
+from github.PullRequest import PullRequest
+from github.Repository import Repository
+
+from consts import OrgFormSchemaIds
+from exceptions import BranchModifiedError
+from parsers import GithubIssueFormDataParser
+
+
+logger = logging.getLogger(__file__)
+
+
+class GitManager:
+    """Manager for creating a new branch and pull request with a file commit in the repo."""
+
+    def __init__(self, repo: Repository):
+        self.repo = repo
+
+    def commit_file_contents_to_branch(
+        self, branch_ref: GitRef, file_path: str, contents: str, commit_message: str
+    ) -> GitCommit:
+        latest_commit = self.repo.get_commit(branch_ref.object.sha)
+        blob = self.repo.create_git_blob(contents, "utf-8")
+        tree_element = InputGitTreeElement(
+            path=file_path, mode="100644", type="blob", sha=blob.sha
+        )
+        base_tree = latest_commit.commit.tree
+        new_tree = self.repo.create_git_tree([tree_element], base_tree)
+        new_commit = self.repo.create_git_commit(
+            message=commit_message, tree=new_tree, parents=[latest_commit.commit]
+        )
+        return new_commit
+
+    def get_or_create_branch(self, source_branch: str, new_branch_name: str) -> GitRef:
+        source = self.repo.get_branch(source_branch)
+        try:
+            branch_ref = self.repo.get_git_ref(f"heads/{new_branch_name}")
+            print(f"Branch '{new_branch_name}' already exists.")
```

Comment:
tutaj chyba coś poszło nie tak bo bezwarunkowo leci ten print. Btw może bardziej by logger pasował :stuck_out_tongue: 

edit: okej, teraz widzę, że `get_git_ref` ma rzucać exception. To może w takim razie ten logger to już poza try..except?

Reply:
Ale jakże? Właśnie o to chodzi, że on loguje, gdy branch już istniał i udało się go pobrać. W innym razie stworzy nowy. Poza try:except zaloguje zawsze, niezależnie czy znalazł, czy stworzył.